docs/proposals/algorithms/lifelong-learning/Towards automatic labeling algorithms for large models of images_reproducing open-world object segmentation based on edge-cloud collaborative lifelong learning.md

```diff
@@ -0,0 +1,101 @@
+# 1 Background
+Traditional machine learning approaches detect objects within known categories by training models with known categories. However, for samples of unknown categories, these models often fail to perform accurate recognition, which limits the performance of the models when tackling open-world tasks. In addition, with the development of large modeling techniques, the need to annotate large-scale data continues to rise, and manually collecting and annotating data becomes a time-consuming and laborious task.
+
+Therefore, open-world object segmentation has become one of the main research directions in the field of Artificial Intelligence. 2023 Segment Anything paper proposed a data engine tool for open-world object segmentation, which can quickly annotate the data and realize the data loop closure. The aim of this project is to reproduce the Segment Anything Model (SAM) model and try to apply the algorithm to open-domain data labeling.
+
+# 2. Goals
+1. reproduce segment anything model based on open world segmentation dataset
+2. achieve >0.45 accuracy for open-world object segmentation (e.g., AP, mIoU).
+3. use the reproduced algorithms to build a data annotation tool, which is integrated into the Sedna and Ianvs lifelong learning modules based on the Ianvs architecture for practical application.
+
+# 3. Proposal
+Open world object segmentation is the core task of this project, and various basic computer vision tasks can be solved by fine-tuning the pre-trained large-scale open world segmentation model **SAM**.
+
+This project reproduces the open world segmentation algorithm **Segment Anything Model** based on sedna and Ianvs lifelong learning, and chooses to fine-tune it on the Autopilot Small Obstacle Detection dataset according to the actual situation. This project tries to use **SAM and its related models** to realize the automatic annotation of open domain data supporting local and cloud deployment, users can send the demanded data to the cloud for processing through the annotation tool at the edge end, and then realize the complete data annotation process through the reasoning module based on SAM and its related algorithms in the cloud.
+
+![Pasted image 20230727172354](https://github.com/winter-fish/ianvs/assets/58922328/f2be38b6-7cb6-4f8b-afeb-c5cfdd0928b4)
+
```

Comment:
According to the given diagram, the user uses the side annotation tool to label the sample, and the annotation tool pushes the sample to the cloud. So how to reflect the automatic annotation, I understand that the automatic annotation should not require the user to participate in the annotation process.